### PR TITLE
[MB-8101] implement new uploads table design

### PR DESCRIPTION
**Pull request by mboren**
## Description

I rewrote the UploadsTable component to match new designs.

### Designs
These designs are for the whole page, not just the component that I'm working on
- [mobile, some uploads](https://app.zeplin.io/project/5f9c266162cd02a552a0b661/screen/609c212513e4c7ab31622465)
- [mobile, no uploads](https://app.zeplin.io/project/5f9c266162cd02a552a0b661/screen/609c2125c29a3abce3128d23)
- [desktop, some uploads](https://app.zeplin.io/project/5f9c266162cd02a552a0b661/screen/609c2126a7cf19ac459cf219)

## Reviewer Notes
Generally I'm not sure what our conventions are for a lot of frontend things.
I was definitely flailing around when writing the CSS, so if you're wondering "did he have a reason for doing it this weird way?" the answer is no.

~I still need to shrink some of the margins on mobile, so some of the text wraps more than it should on smaller screens~ Done.

In the design, when there are no uploads, we don't show the uploads table. I think I've got this working, but I'm still not sure if it's the best way to go about it. ~I tried rendering nothing in this case, but had trouble figuring out how to test that with RTL/jest. I'm also not sure what rendering nothing should look like. Should I return `null`? an empty `div`?~ 

`SectionWrapper` provides styling for `<hr/>` tags that almost matches the lines I need between uploads in the table, but I couldn't figure out a way to stick `<hr>`s between `<li>`s, so I'm using a bottom border instead.

## Setup
Check it out in storybook:
http://localhost:6006/?path=/docs/customer-components-uploadstable--uploads-table

Make sure the upload file size disappears on smaller screens.

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-8101) for this change

### mobile, small
</img>

### mobile, large
</img>

### tablet
</img>

### CODE REVIEW

File: src/components/UploadsTable/UploadsTable.module.scss [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+// should probably rename this
+.wrapper {
+  @include u-padding(3);
+  // section wrapper has its border disappear on mobile,
+  // this doesn't match our design, but idk maybe we want it?
+  border: 1px solid #dcdee0;
+}
+
+h6 {
```
Comment (Shkeating) [inline src/components/UploadsTable/UploadsTable.module.scss:14]: it appears that this is affecting the typography page in storybook?? I'm not sure why. But is you scope the selector to the wrapping class of the component that should fix it.

File: src/components/UploadsTable/UploadsTable.module.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  @include u-margin(0);
+}
+
+ul {
+  @include u-padding-left(0);
```
Comment (Shkeating) [inline src/components/UploadsTable/UploadsTable.module.scss:19]: same here, I am seeing the universal ul styles getting impacted in storybook. Same solution as the previous comment, if you wrap these in your component's wrapping class, that should fix it.

File: src/components/UploadsTable/UploadsTable.module.scss (hunk 1 above)
Comment (mboren) [reply src/components/UploadsTable/UploadsTable.module.scss:14]: Ahh that's probably what's causing the happo diffs. I thought the whole module thing would prevent this, but I guess not.

**Review by jacquelineIO — COMMENTED** [review]
Ran locally fine for me. Looking for a f/e reviewer to approve.

File: src/icons.js [hunk 3]
```diff
@@ -49,6 +49,8 @@ import {
   faTruckMoving,
   faCar,
   faHandHoldingUsd,
+  faFilePdf,
+  faFileImage,
```
Comment (kylehilltruss) [inline src/icons.js:53]: Can you add these to the icons story? `src/stories/icons.stories.jsx`

**Review by kylehilltruss — CHANGES_REQUESTED** [review]
Looks pretty good, other than needing a tweak and adding new icons to the icon story.

File: src/components/UploadsTable/UploadsTable.jsx [hunk 4]
```diff
@@ -1,69 +1,48 @@
...
-        {uploads.map((upload) => (
-          <tr key={upload.id} className="vertical-align text-top">
-            <td className="maxw-card" style={{ overflowWrap: 'break-word', wordWrap: 'break-word' }}>
-              {getUploadUrl(upload)}
-            </td>
-            <td>{moment(upload.created_at).format('LLL')}</td>
-            <td>{bytes(upload.bytes)}</td>
-            <td>
+    uploads.length > 0 && (
+      <SectionWrapper className={classNames(styles.wrapper)}>
```
[the remark this answers is not in the document]
Comment (mboren) [reply src/components/UploadsTable/UploadsTable.jsx]: Done!

File: src/icons.js (hunk 3 above)
Comment (mboren) [reply src/icons.js:53]: Done!

File: src/components/UploadsTable/UploadsTable.module.scss [hunk 5]
```diff
@@ -0,0 +1,75 @@
+@import '../../shared/styles/basics';
+@import '../../shared/styles/mixins';
+@import '../../shared/styles/colors';
+@import '../../shared/styles/_variables';
+
+.wrapper {
```
Comment (jenniferlynparsons) [inline src/components/UploadsTable/UploadsTable.module.scss:6]: This should probably be renamed to something like `uploadsTableContainer`.

**Review by jenniferlynparsons — CHANGES_REQUESTED** [review]
Just one small change for clarity in the naming and otherwise looks good!

File: src/components/UploadsTable/UploadsTable.module.scss (hunk 1 above)
Comment (Shkeating) [reply src/components/UploadsTable/UploadsTable.module.scss]: yeah i thought so too? weird!!

**Review by Shkeating — APPROVED** [review]
looks good!!

File: src/components/UploadsTable/UploadsTable.module.scss (hunk 5 above)
Comment (mboren) [reply src/components/UploadsTable/UploadsTable.module.scss]: Done

**Review by jenniferlynparsons — APPROVED** [review]
Looks good!

**Pull request merged**